### PR TITLE
fix(ci): ask a human once the AI review finishes, not once it goes green

### PR DESCRIPTION
**Pull request by jayantid**
## Summary

`.github/workflows/auto_request_review.yml` assigned a human reviewer only once the `AI Review` check run concluded `success`, which read the bot's conclusions as a verdict they were never meant to be: `success` is its word for "No findings", it never concludes `failure` on purpose, and everything else it has to say — including a review that read the diff and found something — is `neutral`. The gate therefore assigned a reviewer to exactly the pull requests with nothing to look at, and to nothing else. This changes the gate to "the review **finished**" rather than "the review liked the change", leaving two holds that are not judgements on the code: a check that has not completed, and a `conflicted` entry, where the branch does not merge so nothing was read at all.

## Why This Change

Two failure modes, both live on this repository rather than hypothetical.

The first is the ordinary one: a review that finds something concludes `neutral`, so the pull requests that most need a person are the ones the gate silently withheld one from. #890 and #921 both sat with a completed review, findings on the diff, and no reviewer until one was assigned by hand.

The second is worse. Because the gate keyed on green, an outage of the bot became an outage of reviewer assignment — backwards, since a change the automated reviewer could not read is one a person is needed on *more*, not less. Since 2026-08-24 nearly every review here has concluded `Review did not complete` (gke-labs/kube-agents-bot#18), so for the last two days pull requests on this repository have been getting no AI review *and* no human one. Without this change that stays true for as long as the bot's credentials issue does.

## What Changed

- **`scripts/request_reviewers.py`** — `ai_review_block_reason` now clears on any conclusion the bot publishes rather than on `success` alone. Two holds remain: a check that is not `completed`, and an `external_id` of `conflicted`. The conflicted case is read off `external_id`, which the bot documents as its stable machine-readable marker, rather than off the human-readable title. A conclusion the bot does not publish (`failure`, `cancelled`, …) also holds, on the conservative side of a shape nobody has seen — but only for a human author: the `author_is_bot` short circuit sits above that guard, because Dependabot cannot read findings and comment `/review`, so its pull requests pass on *any* completed conclusion, exactly as before this branch.
- **`.github/workflows/auto_request_review.yml`** — the trigger and flag wording follow the gate; `--require-ai-review-pass` is now "once the check has completed".
- **`AGENTS.md`, `docs/site/src/content/docs/contributing.md`** — describe the completed-not-green rule, and name `/review` after a base-branch merge as the way out of a conflicted hold. A push does not start a new review on this repository, so the recovery has to be a command.
- **`scripts/test_request_reviewers.py`** — six new or rewritten gate tests.

## Context

- gke-labs/kube-agents-bot#18 — the credentials outage producing `Review did not complete`, and the reason the gate's behaviour under a broken review is not a corner case right now.
- #890, #921 — the two pull requests that sat unassigned with findings under the old rule.
- **Why here and not in the bot.** The alternative fix is to make the bot conclude `success` when a review completes with findings. That breaks two of its documented invariants — `success` means "read, nothing to raise", and no conclusion is ever a verdict a gate can act on — and it degrades the check's colour as a signal for every other reader in exchange for fixing one consumer. The bot's conclusions are right; the gate was reading them wrong.

## Testing

```
$ uv run --with pytest --with pyyaml python -m pytest scripts/ -q
359 passed, 144 subtests passed in 3.23s

$ uv run --with pytest --with pyyaml python -m pytest scripts/test_request_reviewers.py -q
53 passed, 5 subtests passed in 0.06s
```

- `make docs-check` passes, including the context budget: 36.0k chars, 2.0k under the 38,000 budget.
- The gate tests cover findings, a broken review, a partial review, a conflicted branch, an unpublished conclusion, a running check, a missing check, and the bot-author path on both `neutral` and the five conclusions the bot does not publish.
- **Not run:** `actionlint` and `shellcheck` are not installed on this machine, so `.github/workflows/auto_request_review.yml` is unlinted locally. CI's linters are the check on it. `prettier --check` was likewise not run locally (no Node on this machine); the Markdown edits were kept inside the existing wrap width and `proseWrap` is at its default `preserve`, but CI's prettier job is the real check.

### Live validation

**Not live-tested, and cannot be.** This change touches a GitHub Actions workflow and the script it invokes. It has no path into a kube-agents installation — no CR, no Deployment, no pod — so there is no cluster layer to observe and nothing to take the live-test lease for.

The equivalent evidence is production observation of the bug rather than of the fix: #890 and #921 are the recorded instances of a completed review with findings and no reviewer, and the `Review did not complete` conclusions since 2026-08-24 are the recorded instance of the outage mode. The fix itself is exercised by unit tests over `ai_review_block_reason`, which is where the entire decision lives; the workflow around it only supplies `--require-ai-review-pass` and the check-run payload.

The one thing unit tests cannot prove is that the real `AI Review` check run carries `external_id == "conflicted"` on a conflicted branch. That comes from the bot's documented contract, not from an observed payload on this repository, and it is the assumption most worth a reviewer's eye. If it is wrong, the conflicted hold silently stops holding — a conflicted pull request would get a reviewer assigned early, which is the safe direction to be wrong in.

No test artifacts were created, so none needed cleaning up.

## Self-Review

The branch's original diff was reviewed by a context that did not write it — a fresh session handed the branch and the bot's findings, with none of the reasoning behind the change. Adversarial and docs-drift passes both ran. Stated plainly: the follow-up commit `0aa56a6` was written by that same session, so it has had no independent pass of its own; that is the gap in this section.

- **Ordering bug in the new guard (fixed, `0aa56a6`).** The unpublished-conclusion guard was placed above the `author_is_bot` short circuit, inverting the old order in which `success` returned first and a bot author then passed on any completed conclusion. The effect was that a Dependabot pull request whose check concluded `cancelled`, `timed_out`, `stale` or `skipped` got no reviewer, from an author that cannot type `/review` or `/request-review` — a silent strand, since the run logs a line and exits 0. The narrowing was not intended: the docstring, the AGENTS.md paragraph and the test comment all asserted the opposite. Moved `author_is_bot` back above the guard and added `test_a_conclusion_the_bot_does_not_publish_clears_it_for_a_bot_author`, which sweeps all five conclusions. Raised by the bot; verified against the diff before acting on it.
- **Docs drift on the conflicted recovery (fixed, `0aa56a6`).** Three new passages said a conflicted branch recovers automatically — "fix the conflict and the bot re-runs itself". The repository says the opposite in three places, one of them two bullets above the rewritten line in the same file (`contributing.md` line 145, `resolve_pull_request`'s comment, and AGENTS.md): a push does not start another AI review. So an author following the new text would merge `main`, push, and wait for a check run that is never created. Replaced in all three passages, plus the test comment, with the remedy that works: merge the base branch in, then comment `/review`. Raised by the bot; verified against the three cited passages before acting on it.
- **Does the conflicted hold belong at all?** Checked deliberately, since it is now the only completed outcome that withholds a reviewer from a human. Kept: the bot read nothing, so there is no review for a person to be reading alongside, and the branch does not merge anyway. It is also not a regression — `conflicted` concludes `neutral`, which the old `success`-only gate blocked too. The change is only that the remedy is now documented.
- **Is `external_id` the right key?** Checked against matching on the title copy. Kept `external_id`: the bot documents it as the stable machine-readable marker, and title text is not a contract. Flagged above under Live validation as the assumption this change rests on that tests do not verify.
- **Does the gate still fail closed on the shapes nobody has seen?** Yes for a human author — anything outside `{success, neutral}` holds. Deliberately not for a bot author, per the point above; a bot author is a case where holding strands rather than protects.
- **Looked for and did not find:** any other consumer of `ai_review_block_reason`'s return string (only the workflow log), any other reader of the `AI Review` conclusion in this repository, and any remaining "green"/"passes"/"success" phrasing about the gate in docs after the rewrite.

## Risk & Rollout

Low blast radius and no runtime code paths: this is CI tooling plus its documentation, with nothing shipped in an image or chart. The change is strictly *widening* — every pull request that got a reviewer before still gets one, and some that did not now do — so the new failure mode is a reviewer assigned on a change the bot found something on, which is the intent. `/request-review` and the `agent:ignore` label are unchanged escape hatches in both directions.

Revert is a plain revert of the two commits; nothing persists outside the repository, and no state is migrated. Nothing has to happen at merge time. Worth knowing: the first effect after merge is a burst of reviewer assignments on the open pull requests currently held by the `Review did not complete` outage, as each next `check_run` event re-evaluates them.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: scripts/request_reviewers.py [hunk 1]
```diff
@@ -314,16 +351,21 @@ def ai_review_block_reason(check_run, author_is_bot):
...
+        # Not a conclusion this bot writes, so it is not one this gate knows how
+        # to read. Holding is the conservative direction: the override exists.
+        return f"{AI_REVIEW_CHECK_NAME} concluded {conclusion}, which is not a conclusion it publishes"
 
     if author_is_bot:
-        log(f"{AI_REVIEW_CHECK_NAME} concluded {conclusion}; the author is a bot, so it cannot re-run /review")
         return None
 
+    if check_run.get("external_id") == AI_REVIEW_CONFLICTED_ID:
+        return f"{AI_REVIEW_CHECK_NAME} reports the branch does not merge, so nothing was reviewed"
```
[the remark this answers is not in the document]
Comment (jayantid) [reply scripts/request_reviewers.py:367]: Fixed in 0aa56a6 — you are right, the ordering inverted the old behaviour and nothing in the suite covered it. Moved the `author_is_bot` short circuit back above the `AI_REVIEW_CONCLUSIONS` guard so a bot author passes on any completed conclusion again, and added `test_a_conclusion_the_bot_does_not_publish_clears_it_for_a_bot_author`, which sweeps `cancelled`, `timed_out`, `stale`, `skipped` and `failure`.

File: docs/site/src/content/docs/contributing.md [hunk 2]
```diff
@@ -144,7 +144,7 @@ Every pull request is also reviewed by `kube-agents-bot`, a GitHub App that runs
...
+- **It decides when a human is asked.** The bot posts an `AI Review` check run next to its review — `success` for "No findings", `neutral` when it found something or broke — and a reviewer is auto-assigned once that check **completes**, whichever of those it says. Opening a pull request no longer assigns anyone; the review finishing is what puts your change in front of a person. The exception is `Conflicts with the base branch`, where nothing was read: fix the conflict and the bot re-runs itself. An owner, member, or collaborator can comment `/request-review` to assign one immediately.
 - **Opting out.** The `agent:ignore` label excludes a pull request from review and outranks both commands.
```
[the remark this answers is not in the document]
Comment (jayantid) [reply docs/site/src/content/docs/contributing.md:148]: Fixed in 0aa56a6 — the claim was wrong and contradicted `contributing.md` line 145 two bullets up. All three passages (this line, the `ai_review_block_reason` docstring, the AGENTS.md paragraph) plus the `test_a_conflicted_branch_holds_the_gate` comment now say the way out is to merge the base branch in and comment `/review`, since the push starts no new review.

File: .github/workflows/auto_request_review.yml [hunk 3]
```diff
@@ -1,9 +1,13 @@
 name: Auto Request Review
```
[the remark this answers is not in the document]
Comment (jayantid) [reply .github/workflows/auto_request_review.yml:1]: Rewritten the pull request description against `.github/PULL_REQUEST_TEMPLATE.md`; every section is filled in, including Live validation ("Not live-tested", with why a CI workflow change has no installation to reach and what that leaves unverified) and Self-Review (the passes run, this finding and the other two, and their dispositions).

**Comment by dshnayder** [comment]
Thanks for digging into this — the outage half of this is a real bug and I want it fixed. But I think the change is fixing two different problems with one lever, and only one of them was broken.

## What this changes for a human reviewer

The gate goes from `conclusion == "success"` to "the check reached `completed`". Since the bot reserves `success` for "No findings" / "Not reviewed", the practical effect is that **a PR with unaddressed findings now gets a human auto-assigned.**

That was the property the original gate was bought for. Before it, a reviewer opening a PR knew the author considered it done and the bot had nothing outstanding on it. The loop was: bot finds something → author fixes → `/review` → clean → human. After this change, a reviewer no longer knows which of those states they are walking into. The predictable response is that people batch or defer reviews because they can't tell an incomplete PR from a ready one without opening it — which costs more human time than it saves, and is hard to detect once it starts.

I don't think that cost is discussed in the PR description, and I'd like it weighed before this lands.

## `neutral` is five different states, not one

From `app/review.py` in the bot, everything below concludes `neutral`:

| Title | What happened | Is a human needed? |
| --- | --- | --- |
| `Found N issues` | Review worked, author has work to do | **No** — the author's move |
| `Review did not complete` (`CHECK_BROKEN`) | Bot crashed | **Yes** — nothing else will |
| `No findings in the part that was read` | Clean, but partial coverage | Probably |
| `Review posted` | Output didn't parse | Probably |
| `Conflicts with the base branch` | Nothing was read | No (this PR keeps the hold) |

The evidence in the description — gke-labs/kube-agents-bot#18, and the run of `Review did not complete` since 2026-08-24 — is all row 2. That case is genuinely stuck, and stuck in a way the author cannot escape: the documented remedy is `/review`, which calls the same broken bot. That must be fixed.

Row 1 is not stuck. It has a working remedy the author can execute, and it is the case this PR's widening actually changes for reviewers. Rows 1 and 2 are being fixed together because they share a conclusion string, not because they share a rationale.

## The option that wasn't costed

kube-agents-bot#18 framed this as a binary: either the bot concludes `success` on findings, or the gate treats findings as green. The "Why here and not in the bot" section rebuts the first — correctly, I think; `success` should keep meaning "read, nothing to raise" — and then takes the second.

There's a third option that gets the outage fix without the reviewer-experience cost: **give `CHECK_BROKEN` an `external_id`, the same way `check_conflicted` already has one, and key the gate on that.**

This is exactly the pattern the bot already uses and the pattern this PR already relies on. `CHECK_ID_CONFLICTED` exists because one outcome had to be *recognised* rather than displayed. "The review broke" is a second such outcome — this PR is the consumer that proves it.

**In the bot (`app/review.py`):**

```python
# A second outcome a consumer has to recognise rather than display: the review
# did not run, so its silence is not a verdict on the change.
CHECK_ID_BROKEN = "broken"

CHECK_BROKEN = CheckRun(
    "neutral",
    "Review did not complete",
    "This review broke before it had anything to say, so nothing was posted. "
    f"That is a fault in the reviewer and says nothing about the change. {_NOT_AN_APPROVAL}",
    external_id=CHECK_ID_BROKEN,
)
```

Optionally the same for the partial and unparseable entries (`CHECK_ID_PARTIAL`, `CHECK_ID_UNPARSED`), since a consumer wanting to tell "clean-ish" from "found things" needs those too.

This breaks none of the bot's stated invariants. No conclusion changes, `failure` is still never published, and nothing becomes a merge gate. It only makes an outcome that is already distinct in prose also distinct in machine-readable form. `stamped()` already passes `external_id` through untouched.

**In `scripts/request_reviewers.py`:**

```python
# The conclusions/entries that clear the gate. `success` is a clean read.
# `broken` and `partial` are not verdicts on the change at all — a review the
# bot could not finish is one a person is needed on more, not less.
AI_REVIEW_CLEARING_IDS = {AI_REVIEW_BROKEN_ID, AI_REVIEW_PARTIAL_ID}

if conclusion == "success":
    return None
if check_run.get("external_id") in AI_REVIEW_CLEARING_IDS:
    return None
# Findings hold: the author has a move to make, and making it is what puts
# the change in front of a person.
```

That keeps every fix this PR is actually motivated by, and keeps the property that being assigned a PR means it is ready.

**The honest caveat:** this can't be done in kube-agents alone today. `CHECK_BROKEN` carries `external_id=None`, so distinguishing "broke" from "found issues" right now would mean matching on the title string — which this PR is right to reject as not-a-contract. The bot change has to go first. If the outage needs unblocking before that can ship, I'd rather land this PR as an explicitly temporary unblock with a tracking issue for the narrowing than adopt "completed" as the permanent rule by default.

## Two separate findings on the diff as it stands

**1. The docs edit is wrong about conflict recovery, and the self-review "fixed" it in the wrong direction.**

Three passages were changed to say a push does not restart a conflicted review, so the author must merge the base branch and comment `/review`. The bot says the opposite, and implements it. `check_conflicted`'s docstring:

> The push that resolves the conflicts re-runs this review on its own — the one exception to "pushing is not asking" (`docs/design.md` §10) [...] `CHECK_ID_CONFLICTED` is what makes that retry possible: it is this entry, read back off the previous head commit.

And `app/main.py`'s `_retrying_after_conflict` is that retry — it reads the *previous* head's check run and fires when `external_id == CHECK_ID_CONFLICTED`. The general "pushing does not re-trigger" rule in our docs is real, but conflicted is its documented exception. The original text was right; as written now we'd be telling authors to run a command they don't need. Worth reverting that part in all three passages plus the test comment.

**2. #890 and #921 don't quite support the claim they're cited for.**

Both are offered as PRs that "sat with findings and no reviewer". Both did get human attention: #890 has 7 human reviews from @kyber775, and #921 merged earlier today with an approval. What actually happened on them is that assignment required someone to invoke `/request-review` — which is the documented override working, not a strand. That's a real friction cost and a fair argument for changing something, but it's a weaker claim than "nobody was ever asked to look", and the description leans on the stronger one.

## What I'd suggest

1. File the `CHECK_ID_BROKEN` change on the bot (happy to open it if useful) — small, invariant-preserving, unblocks the outage at the root.
2. Narrow this gate to `success` + the non-verdict ids, keeping the hold on findings.
3. Fix the conflict-recovery passages back to "the push re-runs it" (with `/review` as the impatient path).
4. Soften the #890/#921 claim to what they show: the override was needed, not that no human was reached.

If there's a reason the reviewer-readiness property isn't worth keeping, I'd genuinely like to hear it — but I'd want that argued explicitly rather than arriving as a side effect of the outage fix.

**Comment by dshnayder** [comment]
Coming back to this after thinking harder about it, and I'm walking back part of my earlier comment — the `CHECK_ID_BROKEN` suggestion above. I no longer think the gate should widen at all, and I'd rather we didn't land this in its current shape.

**The rule worth keeping.** A human is assigned only when the AI review completed and found nothing the author needs to address. Everything else — findings, a crash, a conflict, an unparseable result — is the author's to resolve. When a reviewer is assigned, that assignment means the change is ready. That property is the whole value of the gate, and it's cheap to lose and expensive to rebuild, because once reviewers learn that assignment no longer implies readiness they start deferring, and nothing in the system tells you that's happening.

**Nothing is actually stranded under that rule.** Every non-success outcome leaves the author a move: `/review all` for a crash, resolve-and-push for a conflict (the bot re-reads on the post-conflict push), fix-and-`/review` for findings. And `/request-review` is the override — PR authors here are generally MEMBER or COLLABORATOR, so an author who's genuinely stuck can unblock themselves. AGENTS.md already names "a review that never arrived" as one of its cases, in text this PR leaves unchanged.

**Where I now think my earlier reasoning was wrong.** I argued the crash case is special because the author's remedy calls the same broken bot. But bot#18 records #922 concluding `success` between two crashes, so the outage isn't total and retrying is a real move. And a human assigned to a crashed review isn't a second reader — they're the *first* reader of a diff nothing has vetted. That's more human time than the design intends to spend, not less. It's also the reasoning this PR already applies to `conflicted`: "the bot read nothing, so there is no review for a person to be reading alongside." That's equally true of `CHECK_BROKEN`. Applied evenly, the broken review holds too.

Worth adding: unassigned PRs are how bot#18 got noticed at all — "That is how this surfaced — PRs sitting with zero reviewers and all other checks green." This change converts that detector into silence. The crashes keep happening, humans absorb them one PR at a time, and nobody has a reason to chase the root cause.

## But there is a real bug here, and this PR is in the right place to fix it

The author is never told their review broke. `CHECK_BROKEN` posts nothing to the conversation — deliberately, per invariant 5, with the check-run entry as the only record. GitHub doesn't notify on a neutral check-run conclusion, so from the author's side a crashed review is indistinguishable from a slow one until they think to go read the Checks tab. `contributing.md` even documents the discovery method as noticing an absence: "A 👀 with nothing following it 30 minutes later is a bug in the bot."

That's the defect. Not "no reviewer was assigned" — "the author was never told there was anything to do."

And this workflow is already positioned to fix it. It fires on `check_run: [completed]` filtered to `AI Review`, and it already has `issues: write` for the 👀 reaction. It sees every broken and conflicted conclusion today and does nothing but log a line.

**Suggested shape:**

1. **Revert the gate change.** `ai_review_block_reason` goes back to clearing on `success` only.
2. **Notify the author when the bot posted nothing.** Scope it to the two silent paths — `Review did not complete` and `Conflicts with the base branch`. Findings need no notice; the inline comments already notify. Something like:

   > @author — the AI review didn't complete on `abc1234`, so nothing was posted and no reviewer is assigned yet.
   >
   > This is a fault in the reviewer, not your change. Comment `/review all` to retry. If it keeps failing, `/request-review` assigns a human anyway.
   >
   > Known outage: gke-labs/kube-agents-bot#18

3. **Make it idempotent.** The workflow fires on every completed `check_run`, so the notice needs a marker keyed to the head SHA — an HTML comment like `<!-- ai-review-hold: {sha} -->`, list the issue comments, skip if present. The `concurrency` group is already keyed on `head_sha`, which helps but isn't sufficient.
4. **Optionally, a label** (`ai-review-incomplete`) alongside the notice. That restores the fleet-wide detection signal that "zero reviewers assigned" currently provides, in a form you can actually query, without spending a reviewer per crash.

One thing to rule out before someone proposes it: having the workflow post `/review` to self-heal doesn't work. `TRIGGER_ASSOCIATIONS` is `frozenset({"OWNER", "MEMBER", "COLLABORATOR"})`, and the bot also drops comments where `commenter == bot_login`. A `GITHUB_TOKEN` comment is authored by `github-actions[bot]`, whose association falls outside that set — it would be logged and discarded.

**On `external_id`:** with this shape it's needed only to tell the broken entry apart from a findings entry, and today `CHECK_BROKEN` carries `external_id=None`. Adding `CHECK_ID_BROKEN` to the bot is still the clean way to key step 2 — it's the same pattern `CHECK_ID_CONFLICTED` already establishes and it changes no conclusion, so no bot invariant moves. But it's now a small enabler for a notification, not a lever on who gets assigned.

The alternative home for the fix is the bot posting the notice itself, which is arguably where it belongs. That runs into invariant 5, so it's the maintainers' call — and the workflow-side version needs no bot change beyond the optional `external_id`.

The two findings from my earlier comment stand independently: the conflict-recovery docs edit contradicts `_retrying_after_conflict` and should go back, and the #890/#921 citations describe the `/request-review` override being used, not PRs nobody was asked to look at.

**Comment by jayantid** [comment]
Thank you — this is a better read of the problem than the one I shipped, and the `external_id` proposal is the right shape. I'm conceding all three findings. I'm also moving this back to draft, because the root cause turned out to remove the urgency argument this PR was resting on.

## The outage is a credential, not a code path

The diagnostic patch landed in the bot (gke-labs/kube-agents-bot#19) and fired on the `/review` I ran on this PR at 19:29:43Z:

```
ResultError: Claude Code returned an error result: API Error: invalid_grant: Invalid JWT Signature. (exit code: 1)
```

The bot runs Claude Code against Vertex AI with a service account JSON key — `app/review.py:210-224` reads it from `VERTEX_SA_KEY_PATH` into `GOOGLE_APPLICATION_CREDENTIALS`, and `app/settings.py:259-267` mounts it from Secret Manager. The key's private half no longer matches any key registered for that service account. It was rotated or deleted in GCP while the secret kept the old JSON.

So the `CHECK_BROKEN` run that motivated this PR is fixed by rotating a key. Details in gke-labs/kube-agents-bot#18; it needs a project owner, which I am not.

That matters for this PR specifically. You wrote that you'd accept this as an explicitly temporary unblock if the outage needed unblocking before the narrowing could ship. It doesn't. The unblock has a direct fix, so the temporary path has nothing left arguing for it, and I shouldn't take a permanent rule by default.

## Conceding the three findings

**1. The conflict-recovery docs are wrong.** You're right and I verified it. `_retrying_after_conflict` at `app/main.py:313` exists, and its docstring says `docs/design.md` §10 refuses the push trigger "and this is its one exception." The push does re-run the review. I'd have shipped instructions telling authors to run a command they don't need. Reverting all three passages plus the test comment, and this one is wrong independent of everything else here.

**2. The #890 and #921 citations are overstated.** Also verified: #890 has 7 reviews from @kyber775, and you approved #921 yourself at 19:04:17Z. What those PRs show is that `/request-review` had to be invoked — the override working, with a friction cost. Not that nobody was reached. I'll rewrite the claim to what the evidence supports.

**3. The gate lever is too blunt.** `completed` sweeps in "Found N issues", which is the author's move and not a reviewer's, and that does cost the property that being assigned means the PR is ready. I didn't cost that in the description and I should have. Keying on `external_id` gets the outage case without touching the findings case.

## Sequencing

Forced, because `CHECK_BROKEN` carries `external_id=None` today and `CHECK_ID_CONFLICTED` at `app/review.py:1666` is the only one in the file:

1. Rotate the key. Ends the outage now, independent of everything below.
2. Bot: add `CHECK_ID_BROKEN`, and `CHECK_ID_PARTIAL` / `CHECK_ID_UNPARSED` alongside it. **Please do open this if you're still willing** — you have the clearer picture of the invariants and it's your design.
3. This PR, narrowed on top of it: clear the gate on the non-verdict ids, hold it on findings and on conflicts. Docs reverted, claim softened.

Back to draft until step 2 lands.

One thing I'd still argue for keeping, and I think we already agree: a review that broke should put the PR in front of a person. When the bot is down, the documented remedy is `/review`, which calls the same broken bot — the author has no way out. This week is the proof that it happens. That's the part of this PR worth surviving, and your version of it is better than mine.
